src/bmi_cfe.c

```diff
@@ -2116,49 +2128,45 @@ static int Get_value (Bmi *self, const char *name, void *dest)
     return Get_value_at_indices(self, name, dest, inds, 1);
 }
 
-
 static int Set_value_at_indices (Bmi *self, const char *name, int * inds, int len, void *src)
 {
-    if (len < 1)
-        return BMI_FAILURE;
+    void * dest  = NULL;
+    int itemsize = 0;
 
-    // Get "adjusted_index" for variable
-    int adjusted_index = Get_adjusted_index_for_variable(name);
-    if (adjusted_index < 0)
+    if (self->get_value_ptr(self, name, &dest) == BMI_FAILURE)
         return BMI_FAILURE;
 
-    int var_item_size;
-    int status = Get_var_itemsize(self, name, &var_item_size);
-    if (status == BMI_FAILURE)
+    if (self->get_var_itemsize(self, name, &itemsize) == BMI_FAILURE)
         return BMI_FAILURE;
 
-    // For now, all variables are non-array scalar values, with only 1 item of type double
-
-    // Thus, there is only ever one value to return (len must be 1) and it must always be from index 0
-    // ajk: modifying it to work with soil moisture column for rootzone depth based AET
-    if (strcmp(name, "soil_moisture_profile") == 0 || strcmp(name, "soil_layer_depths_m") == 0) { //Adding soil layer depths since they will be needed for root zone adjusted AET estimations -rlm
+    { /* Copy the data */
+      size_t i;
+      size_t offset;
+      char * ptr;
+      for (i=0, ptr=(char*)src; i<len; i++, ptr+=itemsize) {
```

Comment:
A comment here would be quite helpful to describe the pointer manipulation happening.
```suggestion
      // iterate over the source pointer, src, by itemsize byte chunks
      // and set the destination pointer, dest, to the value in src 
      // based on the linear offset provided by inds[i]
      for (i=0, ptr=(char*)src; i<len; i++, ptr+=itemsize) {
```